nodemsgs.go

```diff
@@ -146,13 +146,11 @@ func (n *nodeMsgs) process(outerMsg *pb.Msg) applyable {
 }
 
 func (n *nodeMsgs) next() *pb.Msg {
-	for msg := range n.buffer {
+	for msg := n.buffer.Next(); msg != nil; msg = n.buffer.Next() {
 		switch n.process(msg) {
 		case past:
 			n.oddities.alreadyProcessed(n.id, msg)
-			delete(n.buffer, msg)
 		case current:
-			delete(n.buffer, msg)
 			return msg
 		case future:
 			// TODO, this is too aggressive, but useful for debugging
```

Comment:
In this case, we need to leave the message in the buffer, but it looks like it would always be deleted now?

Reply:
fixed